libraries/AP_BattMonitor/AP_BattMonitor_Analog.cpp

```diff
@@ -7,11 +7,13 @@
 extern const AP_HAL::HAL& hal;
 
 /// Constructor
-AP_BattMonitor_Analog::AP_BattMonitor_Analog(AP_BattMonitor &mon, AP_BattMonitor::BattMonitor_State &mon_state) :
-    AP_BattMonitor_Backend(mon, mon_state)
+AP_BattMonitor_Analog::AP_BattMonitor_Analog(AP_BattMonitor &mon,
+                                             AP_BattMonitor::BattMonitor_State &mon_state,
+                                             AP_BattMonitor_Params &params) :
+    AP_BattMonitor_Backend(mon, mon_state, params)
 {
-    _volt_pin_analog_source = hal.analogin->channel(mon._volt_pin[_state.instance]);
-    _curr_pin_analog_source = hal.analogin->channel(mon._curr_pin[_state.instance]);
+    _volt_pin_analog_source = hal.analogin->channel(_params._volt_pin);
+    _curr_pin_analog_source = hal.analogin->channel(_params._volt_pin);
```

Comment:
I also use "_volt_pin" in "_curr_ pin_analog_source". I think "_curr_ pin".

Reply:
Very good catch, that was a bad use of copy paste, fixed and pushed.